File: solutions/clojure/raindrops/1/src/raindrops.clj
(ns raindrops)

(defn divisible? [num divisor]
  (= 0 (mod num divisor)))

(defn convert
  "Converts a number to its corresponding string of raindrop sounds."
  [num]
  (let [sounds (apply str
                 (cond-> []
                   (divisible? num 3) (conj "Pling")
                   (divisible? num 5) (conj "Plang")
                   (divisible? num 7) (conj "Plong")
                  )
                )
        ]

    (if (empty? sounds)
      (str num)
      sounds)
  )
)

File: solutions/ruby/raindrops/1/raindrops.rb
class Raindrops
  def self.convert(n)
    sound = ""
    sound += "Pling" if (n % 3 == 0)
    sound += "Plang" if (n % 5 == 0)
    sound += "Plong" if (n % 7 == 0)
    
    sound = n.to_s if (sound == "")
    
    return sound
  end
end